lib/streamlit/delta_generator.py

```diff
@@ -435,6 +482,9 @@ def add_rows(self, data=None, **kwargs):
         >>> my_chart.add_rows(some_fancy_name=df2)  # <-- name used as keyword
 
         """
+        # Switch to the active DeltaGenerator, in case we're in a `with` block.
+        self = self._active_dg
```

Comment:
i dont think it hurts to have it here but i'm not quite sure how `with` is supposed to be used here? my natural assumption is a
```
with table:
   st.add_rows()
```
but then we need to add `add_rows` to `init`?

if it's like below, then i don't think we need the `active_dg`?
```
my_table = st.table(df1)
with container:
    my_table.add_rows(df2)
```

Reply:
Hm, that's a really good point that I didn't consider -- how add_rows works with `with`. I initially copied it into here with the guiding principle of "public methods of DeltaGenerator should all set `self = self._active_dg`", to try and head off any accidental cases where we forget to switch DGs.

But on inspection, add_rows isn't currently one of those; we can just stick with `_enqueue`, `_get_coordinates`, and any container-like elements, I think. I'll remove this, and try and think of a better rule for when we need to use active DG.